### PR TITLE
Add support for maistra/istio's maistra-2.1 branch

### PR DESCRIPTION
**Pull request by dgn**
The 2.1 Dockerfile is currently ~just~ mostly a copy of the 2.0 one. I'm currently running the jobs locally to make sure they work, haven't encountered any issues so far. I'll keep a /hold on this PR until the tests have passed.

### CODE REVIEW

**Review by jwendell — COMMENTED** [review]
Can you add it to all repos? (envoy, proxy, operator, etc)

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 HUB ?= quay.io/maistra-dev
 
 BUILD_IMAGE = maistra-builder
-BUILD_IMAGE_VERSIONS = $(BUILD_IMAGE)_2.0 $(BUILD_IMAGE)_1.1 ${BUILD_IMAGE}_1.0
+BUILD_IMAGE_VERSIONS = $(BUILD_IMAGE)_2.1 $(BUILD_IMAGE)_2.0 $(BUILD_IMAGE)_1.1 ${BUILD_IMAGE}_1.0
```
Comment (jwendell) [inline Makefile:4]: Can we drop 1.0?

Comment (dgn) [reply Makefile:4]: different topic, but yes

**Comment by dgn** [comment]
> Can you add it to all repos? (envoy, proxy, operator, etc)

I won't in this PR, but yes we'll do that. I'm focusing on `maistra/istio` for now. It's way more work to do all of them

**Review by jwendell — APPROVED** [review]

**Pull request merged**